server/src/compilerDiagnostics/diagnostics/AddLicenseIdentifier.ts

```diff
@@ -0,0 +1,56 @@
+import { TextDocument } from "vscode-languageserver-textdocument";
+import {
+  CodeAction,
+  CodeActionKind,
+  Diagnostic,
+  Range,
+} from "vscode-languageserver/node";
+import { CompilerDiagnostic, ResolveActionsContext } from "../types";
+import { attemptConstrainToFunctionName } from "../conversions/attemptConstrainToFunctionName";
+import { HardhatCompilerError, ServerState } from "../../types";
+
+const LICENSE_STATEMENT = "// SPDX-License-Identifier: $LICENSE";
+const LICENSES = ["MIT", "GPL-2.0-or-later", "GPL-3.0-or-later", "Unlicense"];
+
+/**
+ * This diagnostic is shown when license identifier is not provided
+ * i.e. SPDX-License-Identifier: MIT;
+ *
+ * The suggested quickfix adds a license identifier on top of the file
+ */
+export class AddLicenseIdentifier implements CompilerDiagnostic {
+  public code = "1878";
+  public blocks: string[] = [];
+
+  public fromHardhatCompilerError(
+    document: TextDocument,
+    error: HardhatCompilerError
+  ): Diagnostic {
+    return attemptConstrainToFunctionName(document, error);
+  }
+
+  public resolveActions(
+    _serverState: ServerState,
+    _diagnostic: Diagnostic,
+    { uri }: ResolveActionsContext
+  ): CodeAction[] {
+    const position = { character: 0, line: 0 };
+
+    return LICENSES.map((license) => ({
+      title: `Add license identifier: ${license}`,
+      kind: CodeActionKind.QuickFix,
+      isPreferred: true,
```

Comment:
I think there should only be one preferred action among a group that can solve the underlying error:

![image](https://user-images.githubusercontent.com/24030/183619248-64583ae6-2273-40be-ab30-c50c48d241d9.png)

In this case I think none `isPreferred`.